### DISCUSSION TITLE
How to automatically logout after the user closed all the tabs

### DISCUSSION THREAD

**Discussion by logandoo** [zauberzeug/nicegui#3288] [Q&A] [answered]
### Question

I am trying to figure out how to validate the login status of users, like when the user closed all the tabs, or after certain time of the user logged in, the app would log out automatically.

I have tried to write  `{"username":"a","authenticated":true}` directly into the storage json file through a sepereate micro-service, but when I called `app.storage.user.get('authenticated')` in the main service, it always returns `True`, no matter the real data in the json file. Any have any good idea?

**Comment by falkoschindler** [comment, marked as answer]
Hi @logandoo,

The storage data is kept in RAM and only written to disk for a potential app restart. Therefore changes to the JSON files are ignored while the app is running.

If you want to make changes to the storage, you can write to `app.storage.user` directly.

**Reply by logandoo** [reply]
ok, thanks

**Comment by me21** [comment]
Shouldn't it be done by setting browser cookie lifetime? When it's expired, the user will be logged off.
